### PR TITLE
Dial back shiny requirement

### PR DESCRIPTION
**Pull request by romainfrancois**
### Intent

addresses #10230 

### Approach

reverts #9847 but then bump requirements induced by `rmarkdown` 2.10: https://github.com/rstudio/rmarkdown/blob/v2.10/DESCRIPTION

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues. 

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

**Comment by romainfrancois** [comment]
I'm not sure how to:

> Enforce that version constraint at runtime only in R Markdown documents using Shiny features (e.g. for runtime: shiny).

**Comment by romainfrancois** [comment]
This commit https://github.com/rstudio/rstudio/commit/9e1d4997464716d2330a6b5aa25e0849fba0afa2 updated `shiny` and related packages, but also `learnr` 

Should this update `learnr` related packages ?

File: src/cpp/session/resources/dependencies/r-packages.json [hunk 1]
```diff
@@ -596,22 +533,8 @@
...
             "packages": [
-                "checkmate",
-                "ellipsis",
-                "evaluate",
-                "fastmap",
-                "htmltools",
-                "htmlwidgets",
-                "jsonlite",
-                "knitr",
                 "learnr",
```
Comment (kevinushey) [inline src/cpp/session/resources/dependencies/r-packages.json:536]: Shouldn't we keep learnr's dependencies here as well?

File: src/cpp/session/resources/dependencies/r-packages.json [hunk 2]
```diff
@@ -605,7 +547,7 @@
                 "jsonlite",
                 "knitr",
                 "learnr",
-                "markdown",
```
Comment (romainfrancois) [inline src/cpp/session/resources/dependencies/r-packages.json:608]: `learnr` says it depends on `rmarkdown` https://github.com/rstudio/learnr/blob/v0.10.1/DESCRIPTION was this a typo ?

File: src/cpp/session/resources/dependencies/r-packages.json (hunk 1 above)
Comment (romainfrancois) [reply src/cpp/session/resources/dependencies/r-packages.json:549]: ✅

File: src/cpp/session/resources/dependencies/r-packages.json (hunk 2 above)
Comment (kevinushey) [reply src/cpp/session/resources/dependencies/r-packages.json:608]: I don't think so, `learnr` tutorials are basically just special R Markdown documents, and so rendering and running those does require the `rmarkdown` package.

File: src/cpp/session/resources/dependencies/r-packages.json [hunk 3]
```diff
@@ -441,27 +390,20 @@
...
-                "glue",
-                "htmltools",
+                "Rcpp",
                 "httpuv",
-                "jsonlite",
-                "later",
-                "lifecycle",
                 "mime",
+                "jsonlite",
+                "xtable",
```
Comment (kevinushey) [inline src/cpp/session/resources/dependencies/r-packages.json:397]: nit: can we keep these alphabetically sorted? (easier to can through + makes diffs a little cleaner)

**Comment by kevinushey** [comment]
LGTM; thanks for putting this together!

**Pull request merged**